lmnts-dev/playwell
# [FEATURE] Course date filtering  #42aeqw

# Overview

Add functionality to allow users to filter courses based on a selected date filter.

## Details
* Install date-fns library to more easily deal with date arithmetic.
* Populate the drop down list for date filter to only show date ranges that are available.
* Update the displayed courses to reflect the selection of the date filter.
* Make sure the drop down list is sorted in correct order (i.e This Month, Next Month, Next Six Months, Next Year)
* Update the URL query parameters to reflect the selected date filter.

## Bonus
* Fixed a bug in age filtering where courses for age < 5 were not being shown.
* Pass in courses filtered by only geo location and category to the age, date, and course filters. (This fix makes it so drop down lists don't inadvertently get filtered based on what courses are visible)

## QA Instructions

* Visit the [programs page](https://5e36462855ef1d0007d13cc6--play-well-staging.netlify.com/programs) on the Playwell staging site.
* Enter in a location.
* Selecting an age filter from the drop down should update the label of the drop down.
* Selecting a date filter from the drop down should update the label of the drop down.
* Selecting a date filter from the drop down should update the list of courses for that date range.
* Selecting a date filter from the drop down should update the course counters below the drop down.
* Selecting a date filter from the drop down should update the url with the query parameters for `date_min` and `date_max`

[#42aeqw](https://app.clickup.com/t/42aeqw)
